src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/apiUsage/JsonEncodingApiUsageInspector.java

```diff
@@ -63,83 +64,121 @@ public PsiElementVisitor buildVisitor(@NotNull final ProblemsHolder holder, bool
             @Override
             public void visitPhpFunctionCall(@NotNull FunctionReference reference) {
                 final String functionName = reference.getName();
-                if (functionName != null) {
-                    if (functionName.equals("json_decode") && this.isFromRootNamespace(reference)) {
-                        final PsiElement[] arguments = reference.getParameters();
-                        if (HARDEN_DECODING_RESULT_TYPE && arguments.length == 1) {
-                            final String replacement = String.format(
-                                    "%sjson_decode(%s, %s)",
-                                    reference.getImmediateNamespaceName(),
-                                    arguments[0].getText(),
-                                    DECODE_AS_ARRAY ? "true" : "false"
-                            );
-                            holder.registerProblem(
-                                    reference,
-                                    MessagesPresentationUtil.prefixWithEa(messageResultType),
-                                    DECODE_AS_ARRAY ? new DecodeIntoArrayFix(replacement) : new DecodeIntoObjectFix(replacement)
-                            );
-                        }
-                        if (HARDEN_ERRORS_HANDLING && arguments.length > 0 && PhpLanguageLevel.get(holder.getProject()).atLeast(PhpLanguageLevel.PHP730)) {
-                            final boolean hasFlag = arguments.length >= 4 && this.hasStricterHandlingFlags(arguments[3]);
-                            if (! hasFlag) {
-                                final String replacement = String.format(
+                if (functionName == null) {
+                    return;
+                }
+
+                if (functionName.equals("json_decode") && this.isFromRootNamespace(reference)) {
+                    final PsiElement[] arguments = reference.getParameters();
+                    if (HARDEN_DECODING_RESULT_TYPE && arguments.length == 1) {
+                        final String replacement = String.format(
+                                "%sjson_decode(%s, %s)",
+                                reference.getImmediateNamespaceName(),
+                                arguments[0].getText(),
+                                DECODE_AS_ARRAY ? "true" : "false"
+                        );
+                        holder.registerProblem(
+                                reference,
+                                MessagesPresentationUtil.prefixWithEa(messageResultType),
+                                DECODE_AS_ARRAY ? new DecodeIntoArrayFix(replacement) : new DecodeIntoObjectFix(replacement)
+                        );
+                    }
+
+                    if (HARDEN_ERRORS_HANDLING && arguments.length > 0 && PhpLanguageLevel.get(holder.getProject()).atLeast(PhpLanguageLevel.PHP730)) {
```

Comment:
I didn't test it, but isn't named arguments a feature that was introduced only in PHP 8.0?

Reply:
Good catch!